### ISSUE TITLE
Debezium: sources for tables created before the start of the consistency topic will return error or wrong results

### ISSUE THREAD

**Issue by philip-stoev** [MaterializeInc/materialize#6555]
### What version of Materialize are you using?

```
materialized v0.7.3-dev (7658bbf64)

```

### What was the issue?

This is a bit difficult to follow, so emphasis added.

The source advances by 1 event for every event mentioned in the ```data_collections``` section of the consistency topic. However, **the consistency topic may have been created after the table was created and populated**, so it will have no ```event_count``` data for inserts/updates made prior.

So if we create a table first:

```
CREATE TABLE t1 (f1 INTEGER PRIMARY KEY);
INSERT INTO t1 VALUES (2);
```

and then configure Debezium, the consistency topic will start like this:

```
{"status":"BEGIN","id":"577","event_count":null,"data_collections":null}
{"status":"END","id":"577","event_count":{"long":0},"data_collections":{"array":[]}}
```

So, Mz will have no rows in the ```mz_source_info``` for ```t1```and this source will not be queryable.

A more interesting thing happens once we perform another insert against t1:

```
INSERT INTO t1 VALUES (2);
```

The consistency topic now reflects this:

```
{"status":"BEGIN","id":"582","event_count":null,"data_collections":null}
{"status":"END","id":"582","event_count":{"long":1},"data_collections":{"array":[{"data_collection":"public.t1","event_count":1}]}}
```

and we now have a row in ```mz_source_info```:

```
materialize=> select * from mz_source_info;
 source_name | source_id | dataflow_id | partition_id | offset | timestamp 
-------------+-----------+-------------+--------------+--------+-----------
 kafka-u1/28 | u1        |          28 | 0            |      1 |         5
```

But the source has not been brought up to speed. Instead, Mz has read **only 1 event from the primary topic**, rather than all the prior events.
```
materialize=> select * from t1;
 f1 
----
  1 <-  this is wrong
(1 row)
```

### Is the issue reproducible? If so, please provide reproduction instructions.

```
$ postgres-execute connection=postgres://postgres:postgres@postgres
CREATE TABLE t1 (f1 INTEGER, PRIMARY KEY (f1));
ALTER TABLE t1 REPLICA IDENTITY FULL;
BEGIN;
INSERT INTO t1 VALUES (1);
INSERT INTO t1 VALUES (2);
INSERT INTO t1 VALUES (3);
COMMIT;

# Give debezium time for the consistency topic to be created
> SELECT mz_internal.mz_sleep(70)
<null>

$ postgres-execute connection=postgres://postgres:postgres@postgres
CREATE TABLE t2 (f1 INTEGER);

> CREATE MATERIALIZED SOURCE t1
  FROM KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'postgres.public.t1'
  WITH (consistency = 'postgres.transaction')
  FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY '${testdrive.schema-registry-url}'
  ENVELOPE DEBEZIUM;

> SELECT * FROM t1;
1
2
3
```

### Please attach any applicable log files.

There are no errors or warnings in the Mz log with this scenario.

**Comment by cuongdo** [comment]
cc @cirego who's looking at this area of the code

**Comment by umanwizard** [comment]
Should be fixed by #6296 according to @cirego
